protos/commands.proto

```diff
@@ -17,49 +17,102 @@ enum Type {
 	ARTIST_COUNT = 8;
 	FOLDER_COUNT = 9;
 	SELECT_QUERY = 10;
-	CONTENT_QUERY = 11;
 	WHEN_DEVICE = 12;
+	CLEAR_DEVICE = 13;
+	SELECT_RESULT = 14;
+	ROW = 15;
+	ONE_INT = 16;
+	ONE_STRING = 17;
+	ONE_FLOAT = 18;
+	DAC_INFO_COMMAND = 19;
+	NEXT_DEVICE = 20;
+}
+
+// This is a return type
+message Row {
+	Type type = 1;
+	map <string, string> results = 2;
+}
+
+// This is a command type
+// It expects a SelectResult in response.
+message DacInfo
+{
```

Comment:
FYI - I believe the style is to have the opening bracket on the same line as the name.